File: API/_Instrument/Service/InstrumentService.cs
using API._Instrument.Dto;
using API._Instrument.Repository;

namespace API._Instrument.Service;

public class InstrumentService(IInstrumentRepository instrumentRepository) : IInstrumentService
{
    public async Task<Guid> CreateInstrumentAsync(CreateInstrumentDto createInstrumentDto, CancellationToken cancellationToken = default)
    {
        return await instrumentRepository.CreateInstrumentAsync(createInstrumentDto, cancellationToken);
    }

    public async Task UpdateInstrumentAsync(UpdateInstrumentDto updateInstrumentDto, CancellationToken cancellationToken = default)
    {
        await instrumentRepository.UpdateInstrumentAsync(updateInstrumentDto, cancellationToken);
    }

    public async Task<List<InstrumentDto>> GetAllInstruments(CancellationToken cancellationToken = default)
    {
        return await instrumentRepository.GetAllInstruments(cancellationToken);
    }

    public async Task<InstrumentDto> GetInstrumentById(Guid instrumentId, CancellationToken cancellationToken = default)
    {
        return await instrumentRepository.GetInstrumentById(instrumentId, cancellationToken);
    }
}

File: Frontend/src/app/_services/instrument/instrument.service.ts
import { inject, Injectable } from '@angular/core';
import {
  CreateInstrumentDto,
  InstrumentApi,
  InstrumentDto,
  UpdateInstrumentDto,
} from '@generated/index';
import { Observable } from 'rxjs';
import { InstrumentServiceInterface } from './instrument.service.interface';

@Injectable({
  providedIn: 'root',
})
export class InstrumentService implements InstrumentServiceInterface {
  #api = inject(InstrumentApi);

  getAllInstruments(): Observable<InstrumentDto[]> {
    return this.#api.apiV1InstrumentGetAllInstrumentsGet();
  }
  getInstrumentById(id: string): Observable<InstrumentDto> {
    return this.#api.apiV1InstrumentGetInstrumentByIdInstrumentIdGet(id);
  }
  createInstrument(instrument: CreateInstrumentDto): Observable<string> {
    return this.#api.apiV1InstrumentCreateInstrumentPost(instrument);
  }
  updateInstrument(instrument: UpdateInstrumentDto): Observable<string> {
    return this.#api.apiV1InstrumentUpdateInstrumentPut(instrument);
  }
}
